recipes/libjxl/meta.yaml

```
@@ -0,0 +1,126 @@
+    sha256: 3114bba1fabb36f6f4adc2632717209aa6f84077bc4e93b420e0d63fa0455c5e
+    patches:
+      # Fix https://github.com/libjxl/libjxl/pull/582
+      - 0001-CHECK_EXCLUDE_LIBS.patch
+      # OpenEXR doesn't have pkg-config files in win64
+      - 0002-OPENEXR_USE_CMAKE.patch  # [win]
+  - git_url: https://skia.googlesource.com/skcms
+    git_rev: 64374756e03700d649f897dbd98c95e78c30c7da
+    folder: third_party/skcms
+
+build:
+  number: 0
+
+requirements:
+  # This section is not recognized by `conda-build`
+  # so deps have to be written in `outputs`.
+  # However, dependencies are not installed for build.sh(bld.bat)
+  # unless the keys are present here.
+  # This is a weird behavior which might be fixed in the future.
+  build: {}
+  host: {}
+
+outputs:
+  - name: libjxl
+    build:
+      run_exports:
+        - {{ pin_subpackage('libjxl', min_pin='x.x', max_pin='x.x') }}
+      ignore_run_exports:
+        - libpng
+        - giflib
+        - openexr
+        - libzlib
+        - libbrotlicommon
+        - libjpeg-turbo
+    requirements:
+      build:
+        - {{ compiler('c') }}  # [not win]
+        - {{ compiler('cxx') }}  # [not win]
+        - clang  # [win]
+        - lld  # [win]
+        - pkg-config
+        - cmake >=3.15
+        - make
+      host:
+        - brotli
+        - libhwy
+        # - lcms2
+        - giflib
+        # Cannot be replaced with `jpeg`
+        # since jpeg_mem_* api is different
+        - libjpeg-turbo
+        - openexr
+        - libpng
+        - zlib  # by libpng
+    script: install.sh  # [not win]
+    script: install.bat  # [win]
+    test:
+      commands:
+        - test -f "$PREFIX"/include/jxl/encode.h  # [not win]
+        - test -f "$PREFIX"/lib/libjxl"$SHLIB_EXT"  # [not win]
+        - test ! -f "$PREFIX"/bin/cjxl  # [not win]
+        - if not exist %LIBRARY_INC%\\jxl\\encode.h exit 1  # [win]
+        - if not exist %LIBRARY_LIB%\\libjxl.dll exit 1  # [win]
+        - if not exist %LIBRARY_BIN%\\cjxl.exe exit 1  # [win]
+
+        # CFEP-18: Static libs shouldn't be part of the main package
+        - test ! -f "$PREFIX"/lib/libjxl.a  # [not win]
+        - if exist %LIBRARY_LIB%\\libjxl-static.lib exit 1  # [win]
+  - name: libjxl-tools
+    build:
+      ignore_run_exports:
+        - libbrotlicommon
+        - libbrotlidec
+        - libzlib
+    requirements:
+      build:
+        - {{ compiler('c') }}  # [not win]
+        - {{ compiler('cxx') }}  # [not win]
+        - clang  # [win]
+        - lld  # [win]
+        - pkg-config
+        - cmake >=3.15
+        - make
+      host:
+        - brotli
+        - libhwy
+        - giflib
+        - libjpeg-turbo
+        - openexr
+        - libpng
+        - zlib
+        - {{ pin_subpackage('libjxl', exact=True) }}
+      run:
+        - libjpeg-turbo
+        - openexr
+        - libpng
+        - {{ pin_subpackage('libjxl', min_pin='x.x', max_pin='x.x') }}
```

```suggestion
        - {{ pin_subpackage('libjxl', exact=True) }}
```

When you pin to subpackages, it is best to pin to exact, instead of loose things to ensure that the view is exactly consistent

---

I see. It looks like a good practice. Thanks!